### PR TITLE
Made inspectdb used Cursor.description.display_size for CharFields' max_length.

### PR DESCRIPTION
**Pull request by felixxm**
`internal_size` is size for fixed-size types not for char types.

### CODE REVIEW

File: django/db/backends/postgresql/introspection.py [hunk 1]
```diff
@@ -113,7 +113,8 @@ def get_table_description(self, cursor, table_name):
             FieldInfo(
                 line.name,
                 line.type_code,
-                line.display_size,
+                # display_size is always None on psycopg2.
```
Comment (felixxm) [inline django/db/backends/postgresql/introspection.py:116]: See [docs](https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.Column.display_size).

**Comment by felixxm** [comment]
buildbot, test on oracle.

File: docs/releases/4.2.txt [hunk 2]
```diff
@@ -407,6 +407,10 @@ Miscellaneous
 * The ``is_summary`` argument of the undocumented ``Query.add_annotation()``
   method is removed.
 
+* :djadmin:`inspectdb` now uses ``display_size`` from
+  ``DatabaseIntrospection.get_table_description()`` rather than
+  ``internal_size``.
```
Comment (Elkasitu) [inline docs/releases/4.2.txt:412]: For char fields?

File: django/db/backends/postgresql/introspection.py [hunk 3]
```diff
@@ -113,7 +113,8 @@ def get_table_description(self, cursor, table_name):
...
+                line.internal_size if line.display_size is None else line.display_size,
```
Comment (Elkasitu) [inline django/db/backends/postgresql/introspection.py:117]: won't this condition always evaluate to `True`?

Comment (felixxm) [reply django/db/backends/postgresql/introspection.py:117]: `display_size` is filled properly on `psycopg`, see ticket-33308.

**Pull request merged**